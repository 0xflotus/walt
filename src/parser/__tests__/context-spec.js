import test from 'ava';
import Context from '../context';

test('syntaxError generates an accurate error string', t => {
  const ctx = new Context({
    token: {
      start: { line: 1, col: 13 },
      end: { line: 1, col: 30 }
    },
    lines: ['let x: i32 = someUnknownToken;'],
    filename: 'test.walt',
    func: { id: 'test' }
  });
  const syntaxError = ctx.syntaxError('Test Error', 'unknown token');

  t.is(syntaxError instanceof SyntaxError, true);
  t.snapshot(syntaxError.toString());
});



