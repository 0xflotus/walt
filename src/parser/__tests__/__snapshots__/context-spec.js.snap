// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`syntaxError generates an accurate error string 1`] = `
"SyntaxError: 
let x: i32 = someUnknownToken;
             ^^^^^^^^^^^^^^^^^ unknown token
Test Error
  at test (test.walt:1:13)"
`;
